sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusMetricsProvider.cs

```diff
@@ -0,0 +1,231 @@
+{
+    internal class ServiceBusMetricsProvider
+    {
+        private const string DeadLetterQueuePath = @"/$DeadLetterQueue";
+
+        private readonly ILogger _logger;
+        private readonly string _entityPath;
+        private readonly ServiceBusEntityType _serviceBusEntityType;
+        private readonly Lazy<ServiceBusReceiver> _receiver;
+        private readonly bool _isListeningOnDeadLetterQueue;
+        private readonly Lazy<ServiceBusAdministrationClient> _administrationClient;
+
+        private DateTime _nextWarningTime;
+
+        public ServiceBusMetricsProvider(
+            string entityPath,
+            ServiceBusEntityType serviceBusEntityType,
+            Lazy<ServiceBusReceiver> receiver,
+            Lazy<ServiceBusAdministrationClient> administrationClient,
+            ILoggerFactory loggerFactory)
+        {
+            _serviceBusEntityType = serviceBusEntityType;
+            _receiver = receiver;
+            _entityPath = entityPath;
+            _isListeningOnDeadLetterQueue = entityPath.EndsWith(DeadLetterQueuePath, StringComparison.OrdinalIgnoreCase);
+            _administrationClient = administrationClient;
+            _logger = loggerFactory.CreateLogger<ServiceBusMetricsProvider>();
+            _nextWarningTime = DateTime.UtcNow;
+        }
+
+        public async Task<long> GetMessageCount()
+        {
+            long activeMessageCount = 0, deadLetterCount = 0;
+            string entityName = _serviceBusEntityType == ServiceBusEntityType.Queue ? "queue" : "topic";
+            try
+            {
+                if (_serviceBusEntityType == ServiceBusEntityType.Queue)
+                {
+                    QueueRuntimeProperties queueRuntimeProperties = await _administrationClient.Value.GetQueueRuntimePropertiesAsync(_entityPath).ConfigureAwait(false);
+                    activeMessageCount = queueRuntimeProperties.ActiveMessageCount;
+                    deadLetterCount = queueRuntimeProperties.DeadLetterMessageCount;
+                }
+                else
+                {
+                    ServiceBusEntityPathHelper.ParseTopicAndSubscription(_entityPath, out string topicPath, out string subscriptionPath);
+
+                    SubscriptionRuntimeProperties subscriptionProperties = await _administrationClient.Value.GetSubscriptionRuntimePropertiesAsync(topicPath, subscriptionPath).ConfigureAwait(false);
+                    activeMessageCount = subscriptionProperties.ActiveMessageCount;
+                    deadLetterCount = subscriptionProperties.DeadLetterMessageCount;
+                }
+            }
+            catch (UnauthorizedAccessException ex) // When manage claim is not used on Service Bus connection string
+            {
+                string message = $"Connection string does not have 'Manage Claim' for {entityName} '{_entityPath}'. Target base scale won't work due impossibility to get active message count.";
```

Comment:
Remove TBS specific wording here since this is a general method. E.g. "Connection string does not have 'Manage Claim' for {entityName} '{_entityPath}'. Unable to determine active message count."